iml-services/iml-snapshot/src/main.rs

```diff
@@ -2,16 +2,128 @@
 // Use of this source code is governed by a MIT-style
 // license that can be found in the LICENSE file.
 
+use core::fmt::Debug;
 use futures_util::stream::TryStreamExt;
+use iml_command_utils::{wait_for_cmds_success, CmdUtilError};
+use iml_graphql_queries::snapshot as snapshot_queries;
+use iml_manager_client::{get_influx, graphql, ImlManagerClientError};
 use iml_manager_env::get_pool_limit;
 use iml_postgres::{get_db_pool, sqlx};
 use iml_service_queue::service_queue::consume_data;
 use iml_tracing::tracing;
 use iml_wire_types::snapshot;
+use reqwest::Client;
+use std::{collections::HashMap, collections::HashSet, sync::Arc};
+use thiserror::Error;
+use tokio::{
+    sync::Mutex,
+    time::Instant,
+    time::{interval, Duration},
+};
 
 // Default pool limit if not overridden by POOL_LIMIT
 const DEFAULT_POOL_LIMIT: u32 = 2;
 
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+struct SnapshotId {
+    filesystem_name: String,
+    snapshot_name: String,
+    snapshot_fsname: String,
+}
+
+#[derive(Error, Debug)]
+enum Error {
+    #[error(transparent)]
+    CmdUtilError(#[from] CmdUtilError),
+    #[error(transparent)]
+    ReqwestError(#[from] reqwest::Error),
+    #[error(transparent)]
+    ImlManagerClientError(#[from] ImlManagerClientError),
+    #[error(transparent)]
+    ImlGraphqlQueriesError(#[from] iml_graphql_queries::Errors),
+}
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+enum State {
+    Monitoring(u64),
+    CountingDown(Instant),
+}
```

Comment:
A snapshot is monitoring or being monitored?

Reply:
Me (the service) is monitoring the state of a snapshot